tests/test_publisher.py

```diff
@@ -9,11 +11,9 @@ def test_Command_triggers_no_warnings(recwarn):
     with Command(["echo"]) as client:
         for _ in client:
             pass
-    # Delete our reference so that the Command instance gets garbage
-    # collected here. Otherwise, gc will not happen until after the
-    # test completes and warnings emitted during gc will not be captured
-    # by the recwarn fixture.
-    del client
+    # Check for warnings during garbage collection, too
+    client = weakref.ref(client)
```

Comment:
Maybe you want to explicitly run garbage collection here? https://docs.python.org/3/library/gc.html?highlight=gc#gc.collect

Reply:
I don't think it is necessary.  Objects finalized as soon as their reference count gets to zero.   Explicitly triggering garbage collection should only be necessary when there are reference cycles involved.  As `gc.collect()` triggers a full mark-and-sweep collection of all live objects (I think), it can take a bit of time to run. If there's no reason to do it, I think it's best to avoid.

I just added the weakref sanity check just to make sure that the instance is actually being finalized when we think it is.

Reply:
Yes, this is true for CPython (but I don't think it's true for some other implementations like PyPy). It'd make the test a little bit more explicit about what we're testing and it would be future proof if we decide to officially support other interpreters. It's nothing major though.

Reply:
You're right — [PyPy does not](https://doc.pypy.org/en/latest/cpython_differences.html#differences-related-to-garbage-collection-strategies) rely on reference counting to trigger finalization.

I've updated the test again.  See if you like it.

Reply:
Looks good!